### ISSUE TITLE
Question about Neural Network normalization

### ISSUE THREAD

**Issue by lindapaiste** [ml5js/ml5-next-gen#96]
I'm looking at the code for preparing inputs for the Neural Network.  We always convert non-numeric values to number arrays (strings through one-hot encoding and images to pixel arrays). 

I'm confused about when we handle normalization of numbers to the range `[0, 1]` based on the `min` and `max`.  It appears that this is an optional step which only happens if the user has called the `normalizeData()` function. 

What I'm seeing in the code is that calling `normalizeData()` sets the `this.neuralNetworkData.meta.isNormalized` flag to `true`.  All calls to the normalize functions are guarded inside `if (meta.isNormalized) {` blocks.  So if `normalizeData()`  has not been called by the user at the top level then no numbers will be normalized.

@shiffman is that correct?  Is that how it's supposed to work?  What are the situations where someone might chose to normalize the data or not?  This is definitely something that we should explain in the docs.

**Comment by lindapaiste** [comment]
We could possibly handle this through the options, with normalization enabled by default and disabled with `{ normalize: false }`. Perhaps we could accept the range `{ normalize: [-1, 1] }` and have `[0, 1]` as the default.

It's expected that all input fields should be normalized to the same range, correct?  So if the range is `[-1, 1]` then our one-hot encoded strings would be `[1, -1, -1, -1]` etc.  That can be achieved either by mapping on our end or by using the [`onValue` and `offValue` arguments of tf.oneHot](https://js.tensorflow.org/api/latest/#oneHot).
